src/engine/renderer/tr_bsp.cpp

```diff
@@ -955,7 +955,7 @@ static void ParseFace( dsurface_t *ds, drawVert_t *verts, bspSurface_t *surf, in
 	for( i = 0; i < numVerts; i++ ) {
```

Comment:
`rintf` is kinda icky since it requires some archaic floating point coprocessor behaviors. `std::roundf` would be nicer

Reply:
> There's another copy of this code on line 1273; do we also need to change that?

I guess that's a question for @gimhael :grin:

> `rintf` is kinda icky since it requires some archaic floating point coprocessor behaviors. `std::roundf` would be nicer

[looks](https://en.cppreference.com/w/cpp/numeric/math/round) to be usable as a [drop-in](https://en.cppreference.com/w/c/numeric/math/rint) replacement, I see nothing against such change, let's do it if it's better.

Note: can you comment _under_ the line? Github is not really friendly with quoting above the line.

Reply:
Yes, the code in line 1273 should be changed too. This is the loader for triangle meshes, there are probably no rotating textures on triangle meshes, but it's better to keep it consistent.

The rintf uses the current rounding mode while roundf guarantees round to nearest, which implies that roundf may change the rounding mode. This is a very expensive operation on x86, thus I preferred rintf() here.

Reply:
We have SSE instructions enabled on x86 so I was hoping the x87 stuff could be sidestepped (I'm assuming this slow mode problem doesn't exist with SSE but I could also be wrong about that...), but I suppose library functions probably don't take advantage of SSE, so gimhael's point is correct. At least, if we want to continue to treat x86 as a first class citizen, which may be a questionable decision...

But anyway, there's a nice way we can sidestep the rounding mode nonsense while also (maybe) optimizing the code. Instead of `round(x - 0.5)` we can do `floor(x)`. (using `std::floorf`)

Reply:
Surprisingly to me, rintf is [apparently](https://godbolt.org/z/Ex2SPF) a builtin in gcc (so is floorf, but not roundf). And rounding x - 0.5 actually generates fewer instructions. So it looks like the original version is pretty great. Thanks for the lesson @gimhael :)